lib/datadog/core/telemetry/metrics_manager.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+require_relative 'metrics_collection'
+
+module Datadog
+  module Core
+    module Telemetry
+      # MetricsManager aggregates and flushes metrics and distributions
+      class MetricsManager
+        attr_reader :enabled
+
+        def initialize(aggregation_interval:, enabled:)
+          @interval = aggregation_interval
+          @enabled = enabled
+          @mutex = Mutex.new
+
+          @collections = {}
+        end
```

Comment:
We don't create namespaces dynamically, but also they are managed outside of this library in internal telemetry metrics definition file: 

https://github.com/DataDog/dd-go/blob/prod/trace/apps/tracer-telemetry-intake/telemetry-metrics/static/common_metrics.json

The namespaces examples are: 
- general
- tracers
- civisibility

I did not want to duplicate this information in this library, so I went with a way to just start reporting metrics for any new namespace that might be created.

Reply:
I see. :thinking: seeing there's only 3 makes me think we could list them on our side as well and be done with it -- thus avoiding the locking on the manager class (and the overhead of it, including in the testing with multiple threads).

We could add a `raise` if a namespace is unknown, so any lib developers would immediately know they need to update the list if they want to start submitting to the new namespace.

Having said that, I'll stand by what I said on my approval note -- I'd go with the simplification, but I leave the final decision to you!

Reply:
no, sorry, there are more than 3, I was just lazy to copy paste them all

Reply:
there is also:
- appsec
- iast
- profilers

So 6 overall, also not many. I would like to keep it as is, because js and python also use similar approach and I really don't want to support the list of namespaces on our side.